crates/primitives/src/result.rs

```diff
@@ -16,7 +16,6 @@ pub type EVMResultGeneric<T, ChainSpecT, DBError> = core::result::Result<
 >;
 
 #[derive(Debug, Clone, PartialEq, Eq)]
-#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
```

Comment:
Removing serde (de)serialisation allowed the removal of the serde-related trait bounds from the `HaltReasonTrait`. As the `ExecutionResult` and `ResultAndState` aren't part of the JSON-RPC of Ethereum, I think it should be safe to remove this.

WDYT?